lib/msal-common/src/logger/Logger.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { StringUtils } from "../utils/StringUtils";
+import pkg from "../../package.json";
+
+export type LoggerMessageOptions = {
+    logLevel: LogLevel,
+    correlationId?: string,
+    containsPii?: boolean
+};
+
+export enum LogLevel {
+    Error,
+    Warning,
+    Info,
+    Verbose
+};
+
+export interface ILoggerCallback {
+    (level: LogLevel, message: string, containsPii: boolean): void;
+}
+
+export class Logger {
+
+    /**
+     * @hidden
+     */
+    private correlationId: string;
+
+    /**
+     * @hidden
+     */
+    private level: LogLevel = LogLevel.Info;
+
+    /**
+     * @hidden
+     */
+    private piiLoggingEnabled: boolean;
+
+    /**
+     * @hidden
+     */
+    private localCallback: ILoggerCallback;
+
+    constructor(localCallback: ILoggerCallback, piiLoggingEnabled: boolean) {
+        this.localCallback = localCallback;
+        this.piiLoggingEnabled = piiLoggingEnabled;
+    }
+
+    /**
+     * @hidden
+     */
+    private logMessage(logMessage: string, options: LoggerMessageOptions): void {
+        if ((options.logLevel > this.level) || (!this.piiLoggingEnabled && options.containsPii)) {
+            return;
+        }
+        const timestamp = new Date().toUTCString();
+        let log: string;
+        if (!StringUtils.isEmpty(this.correlationId)) {
+            log = timestamp + ":" + this.correlationId + "-" + pkg.version + "-" + LogLevel[options.logLevel] + " " + logMessage;
```

Comment:
I have many thoughts on this function. Let's chat tomorrow.

Reply:
Can this part be changed to a ternary, with string templates instead of manual concatenation?

Reply:
Also, some more whitespace in teh messages would be nice.

Reply:
will make some changes.

Reply:
Made some changes, please review again.